hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapReplicationStateHolder.java

```diff
@@ -167,6 +167,9 @@ void applyState() {
 
                 long nowInMillis = Clock.currentTimeMillis();
                 final InternalIndex[] indexesSnapshot = indexes.getIndexes();
+
+                Indexes.beginPartitionUpdate(indexesSnapshot);
```

Comment:
I do not think this is needed. When `clear` is called, the `PartitionIdSet` of indexed partitions is cleared. Therefore, if the query expected some partitions to be indexed, say `[1, 2, 3, 4]`, this will not be the case, and the query will fail instead of returning incorrect results.